### ISSUE TITLE
checkOutput throws NullPointerException

### ISSUE THREAD

**Issue by cranst0n** [tpolecat/doobie#262]
Using MySQL, running the following query works within the MySQL console and also via JDBC:

``` scala
private[db] def statesForTimeRangeQ(timeStart: Instant, timeEnd: Instant) = {
    sql"""
      (select
        s.unixTime, s.sampleRate, s.zoneSpacing, s.zoneOffset, s.frameTime, s.numZones, s.firstZone
      from SystemState s where s.unixTime < $timeStart
        order by s.unixTime desc limit 1)
      union
      (select
        s.unixTime, s.sampleRate, s.zoneSpacing, s.zoneOffset, s.frameTime, s.numZones, s.firstZone
      from SystemState s where s.unixTime between $timeStart and $timeEnd
        order by s.unixTime desc)
    """.query[SystemState]
  }
```

This Specs2 fragment passes as expected:

``` scala
val sampleSystemState = SystemState(Instant.now(), 12.3f, 34.5f, 678, 9.0f, 1234, 1)
statesForTimeRangeQ(
  sampleSystemState.time.minusMillis(1000),
  sampleSystemState.time.plusMillis(1000)
).list.transact(xa).run must contain(exactly(sampleSystemState))
```

However this:

``` scala
checkOutput(
  statesForTimeRangeQ(
    sampleSystemState.time.minusMillis(1000),
    sampleSystemState.time.plusMillis(1000)
  )
)
```

throws:

``` sbt
[error]       ! SQL Compiles and Typechecks
[error]        java.lang.NullPointerException: null (pretty.scala:58)
[error] doobie.util.pretty$.wrap(pretty.scala:58)
[error] doobie.contrib.specs2.analysisspec$AnalysisSpec$class.doobie$contrib$specs2$analysisspec$AnalysisSpec$$formatError(specs2.scala:87)
[error] doobie.contrib.specs2.analysisspec$AnalysisSpec$$anonfun$checkAnalysis$3$$anonfun$apply$1.apply(specs2.scala:64)
[error] doobie.contrib.specs2.analysisspec$AnalysisSpec$$anonfun$checkAnalysis$3$$anonfun$apply$1.apply(specs2.scala:64)
```

**Comment by tpolecat** [comment]
It looks like an exception is being raised during analysis but `.getMessage` is returning `null`, which is breaking the output formatter. If you can try the following in the REPL it might provide a clue.

```scala
statesForTimeRangeQ(
  sampleSystemState.time.minusMillis(1000),
  sampleSystemState.time.plusMillis(1000)
).analysis.transact(xa).run
```

My expectation is that it will throw an exception that might provide more clues.

**Comment by cranst0n** [comment]
Here's what I get:

```sbt
[info] SystemStateComponent should
[error]   ! have typechecked queries
[error]    java.sql.SQLException: Parameter metadata not available for the given statement (SQLError.java:957)
[error] com.mysql.jdbc.SQLError.createSQLException(SQLError.java:957)
[error] com.mysql.jdbc.SQLError.createSQLException(SQLError.java:896)
[error] com.mysql.jdbc.SQLError.createSQLException(SQLError.java:885)
[error] com.mysql.jdbc.SQLError.createSQLException(SQLError.java:860)
[error] com.mysql.jdbc.MysqlParameterMetadata.checkAvailable(MysqlParameterMetadata.java:69)
[error] com.mysql.jdbc.MysqlParameterMetadata.getParameterType(MysqlParameterMetadata.java:117)
[error] doobie.hi.preparedstatement$$anonfun$getParameterJdbcMeta$1$$anonfun$apply$3.apply(preparedstatement.scala:159)
[error] doobie.hi.preparedstatement$$anonfun$getParameterJdbcMeta$1$$anonfun$apply$3.apply(preparedstatement.scala:158)
[error] doobie.hi.preparedstatement$$anonfun$getParameterJdbcMeta$1.apply(preparedstatement.scala:158)
[error] doobie.hi.preparedstatement$$anonfun$getParameterJdbcMeta$1.apply(preparedstatement.scala:157)
[error] foss.db.SystemStateComponentSpec$$anonfun$1$$anonfun$apply$2.apply(SystemStateComponentSpec.scala:40)
[error] foss.db.SystemStateComponentSpec$$anonfun$1$$anonfun$apply$2.apply(SystemStateComponentSpec.scala:30)
```

**Comment by tpolecat** [comment]
Ah, sorry. MySQL. Can you try again with `outputAnalysis` rather than `analysis`?

**Comment by cranst0n** [comment]
```scala
[error]   ! have typechecked queries
[error]    java.lang.NullPointerException: null (preparedstatement.scala:119)
[error] doobie.hi.preparedstatement$$anonfun$getColumnJdbcMeta$1.apply(preparedstatement.scala:119)
[error] doobie.hi.preparedstatement$$anonfun$getColumnJdbcMeta$1.apply(preparedstatement.scala:118)
[error] foss.db.SystemStateComponentSpec$$anonfun$1$$anonfun$apply$2.apply(SystemStateComponentSpec.scala:40)
[error] foss.db.SystemStateComponentSpec$$anonfun$1$$anonfun$apply$2.apply(SystemStateComponentSpec.scala:30)
```

**Comment by tpolecat** [comment]
Ok the exception is here ... it looks like the driver is returning `null` when we ask for column metadata on the `ResultSet` (the NPE arises from `md.getColumnCount`).

```scala
  def getColumnJdbcMeta: PreparedStatementIO[List[ColumnMeta]] =
    PS.getMetaData.map { md =>
      (1 to md.getColumnCount).toList.map { i =>
        ...
```

I think the TL;DR is that the driver can't provide any useful information about this query. But doobie should be able to cope with `null` here so I'll fix that bit.

**Comment by cranst0n** [comment]
Fair enough. Thanks for looking into this so quickly.

**Comment by fernandomora** [comment]
Is this fixed either in 0.2.x or 0.3.x branch?
